code/modules/clothing/masks/gasmask.dm

```diff
@@ -45,15 +54,73 @@ GLOBAL_LIST_INIT(clown_mask_options, list(
 	QDEL_LAZYLIST(gas_filters)
 	return..()
 
+/obj/item/clothing/mask/gas/equipped(mob/equipee, slot)
+	if(cig)
+		cig.equipped(equipee, slot)
+	return ..()
+
+/obj/item/clothing/mask/gas/adjustmask(mob/living/carbon/user)
+	if(cig)
+		user.visible_message(
+				span_warning("You need to remove [cig] before you can adjust the mask")
+		)
+		return
+	return ..()
+
+
 /obj/item/clothing/mask/gas/examine(mob/user)
 	. = ..()
-	if(max_filters > 0)
+	if(cig)
+		. += "<span class='notice'>There is a [cig.name] jammed into the filter slot</span>"
+	if(max_filters > 0 && !cig)
 		. += "<span class='notice'>[src] has [max_filters] slot\s for filters.</span>"
 	if(LAZYLEN(gas_filters) > 0)
 		. += "<span class='notice'>Currently there [LAZYLEN(gas_filters) == 1 ? "is" : "are"] [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.</span>"
 		. += "<span class='notice'>The filters can be removed by right-clicking with an empty hand on [src].</span>"
 
+/obj/item/clothing/mask/gas/proc/on_cig_destroyed()
+	SIGNAL_HANDLER
+	if(cig)
+		cig = null
+		if(istype(src.loc, /mob))
+			var/mob/wearer = src.loc
+			wearer.update_worn_mask()
+
 /obj/item/clothing/mask/gas/attackby(obj/item/tool, mob/user)
+	if(istype(tool, /obj/item/clothing/mask/cigarette))
+		var/valid_wearer = istype(src.loc, /mob)
+
+		if(flags_cover & MASKCOVERSMOUTH)
+			user.visible_message(
+				span_warning("You can't do that while the mask's mouth is covered")
+			)
+			return ..()
+
+		if(max_filters <= 0 || cig)
+			user.visible_message(
+				span_warning("There's nowhere for that to fit")
+			)
+			return ..()
+		if(has_filter)
+			user.visible_message(
+				span_warning("You can't fit that in while there are filters in the mask")
+			)
```

Comment:
these would all be better as balloon alerts. I'm also a big fan of multilining, but it doesn't make any sense when there's only one entry in the list, just looks uglier.